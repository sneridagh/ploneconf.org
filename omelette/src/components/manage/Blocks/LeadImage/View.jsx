/**
 * View image block.
 * @module components/manage/Blocks/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import config from '@plone/volto/registry';

import { flattenToAppURL } from '@plone/volto/helpers';

/**
 * View image block class.
 * @class View
 * @extends Component
 */
const View = ({ data, properties }) => (
  <p
    className={cx(
      'block image align',
      {
        center: !Boolean(data.align),
      },
      data.align,
    )}
  >
    {properties.image && (
      <>
        {(() => {
          const image = (
            <img
              className={cx({ 'full-width': data.align === 'full' })}
              src={flattenToAppURL(properties.image.download)}
              alt={properties.image_caption || ''}
            />
          );
          if (data.href) {
            if (
              (data.href.startsWith('http') || data.href.startsWith('https')) &&
              !data.href.includes(config.settings.apiPath)
            ) {
              return (
                <a
                  target={data.openLinkInNewTab ? '_blank' : null}
                  href={data.href}
                >
                  {image}
                </a>
              );
            } else {
              return (
                <Link
                  to={data.href.replace(config.settings.apiPath, '')}
                  target={data.openLinkInNewTab ? '_blank' : null}
                >
                  {image}
                </Link>
              );
            }
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  properties: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
