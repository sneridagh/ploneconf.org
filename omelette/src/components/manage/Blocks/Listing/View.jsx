import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { ListingBlockBody as ListingBody } from '@plone/volto/components';

const View = ({ data, properties, block, path }) => {
  return (
    <div className={cx('block listing', data.template)}>
      <ListingBody
        data={data}
        properties={properties}
        block={block}
        path={path}
      />
    </div>
  );
};

View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  properties: PropTypes.objectOf(PropTypes.any).isRequired,
  block: PropTypes.string,
};

export default View;
