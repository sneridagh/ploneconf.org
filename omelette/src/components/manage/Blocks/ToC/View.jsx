/**
 * View toc block.
 * @module components/manage/Blocks/ToC/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { map, filter } from 'lodash';
import { List } from 'semantic-ui-react';
import { defineMessages, injectIntl } from 'react-intl';
import AnchorLink from 'react-anchor-link-smooth-scroll';

import {
  getBlocksFieldname,
  getBlocksLayoutFieldname,
} from '@plone/volto/helpers';

const messages = defineMessages({
  text: {
    id: 'Table of Contents',
    defaultMessage: 'Table of Contents',
  },
});

/**
 * View toc block class.
 * @class View
 * @extends Component
 */
const View = ({ properties, data, intl }) => {
  const blocksFieldname = getBlocksFieldname(properties);
  const blocksLayoutFieldname = getBlocksLayoutFieldname(properties);
  const placeholder = data?.placeholder || intl.formatMessage(messages.text);
  return (
    <div className="block table-of-contents">
      <h2>{placeholder}</h2>
      <List bulleted>
        {map(
          filter(
            map(
              properties[blocksLayoutFieldname].items,
              (id) => properties[blocksFieldname][id],
            ),
            (block) =>
              block['@type'] === 'text' &&
              block.text?.blocks[0].type.indexOf('header-') === 0,
          ),
          (block) => (
            <List.Item
              key={block.text.blocks[0].key}
              className={block.text.blocks[0].type}
            >
              <AnchorLink href={`#${block.text.blocks[0].key}`}>
                {block.text.blocks[0].text}
              </AnchorLink>
            </List.Item>
          ),
        )}
      </List>
    </div>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  properties: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default React.memo(injectIntl(View));
