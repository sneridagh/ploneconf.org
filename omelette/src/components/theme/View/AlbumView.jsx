/**
 * Album view component.
 * @module components/theme/View/AlbumView
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Container, Image, GridColumn, Segment } from 'semantic-ui-react';
import { Button, Modal, Grid } from 'semantic-ui-react';
import { Icon } from '@plone/volto/components';

import openSVG from '@plone/volto/icons/open.svg';
import aheadSVG from '@plone/volto/icons/ahead.svg';
import backSVG from '@plone/volto/icons/back.svg';

import { flattenToAppURL } from '@plone/volto/helpers';

/**
 * Album view component class.
 * @function AlbumView
 * @param {Object} content Content object.
 * @returns {string} Markup of the component.
 */
class AlbumView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      openIndex: undefined,
    };

    this.closeModal = this.closeModal.bind(this);
    this.nextImage = this.nextImage.bind(this);
    this.prevImage = this.prevImage.bind(this);
  }

  closeModal() {
    this.setState({
      openIndex: -1,
    });
  }

  nextImage() {
    const openIndex =
      (this.state.openIndex + 1) % this.props.content.items.length;
    this.setState({
      openIndex,
    });
  }

  prevImage() {
    const openIndex =
      (this.state.openIndex - 1) % this.props.content.items.length;
    this.setState({
      openIndex,
    });
  }

  render() {
    const { content } = this.props;
    return (
      <Container className="view-wrapper">
        <article id="content">
          <header>
            <h1 className="documentFirstHeading">{content.title}</h1>
            {content.description && (
              <p className="documentDescription">{content.description}</p>
            )}
          </header>
          <section id="content-core">
            <Grid doubling stackable columns={4}>
              {content.items &&
                content.items.map((item, index) => (
                  <React.Fragment key={item.url}>
                    {item.image && (
                      <Modal
                        className="gallery"
                        onClose={this.closeModal}
                        open={this.state.openIndex === index}
                        trigger={
                          <Grid.Column>
                            <Segment className="imageborder">
                              <Image
                                verticalAlign="middle"
                                alt={
                                  item.image_caption
                                    ? item.image_caption
                                    : item.title
                                }
                                src={flattenToAppURL(
                                  item.image.scales.preview.download,
                                )}
                                onClick={() => {
                                  this.setState({
                                    openIndex: index,
                                  });
                                }}
                              />
                            </Segment>
                          </Grid.Column>
                        }
                        closeIcon
                      >
                        <Modal.Header>
                          <Grid>
                            <Grid.Row>
                              <GridColumn width={10}>{item.title}</GridColumn>
                              <GridColumn width={2} textAlign="right">
                                <Link
                                  to={item.url}
                                  title={item['@type']}
                                  onClick={this.closeModal}
                                >
                                  <Icon size="30px" fitted name={openSVG} />
                                </Link>
                              </GridColumn>
                            </Grid.Row>
                          </Grid>
                        </Modal.Header>
                        <Grid centered verticalAlign="middle">
                          <Grid.Row>
                            <Grid.Column width={2} textAlign="center">
                              <Button
                                className="gallery noborder"
                                onClick={this.nextImage}
                                style={{ margin: 0 }}
                              >
                                <Icon
                                  name={backSVG}
                                  className="circled"
                                  size="30px"
                                  style={{ margin: 0 }}
                                />
                              </Button>
                            </Grid.Column>
                            <Grid.Column width={8}>
                              <Modal.Content image>
                                <Image
                                  wrapped
                                  alt={
                                    item.image_caption
                                      ? item.image_caption
                                      : item.title
                                  }
                                  src={item.image.scales.large.download}
                                />
                                <Modal.Description>
                                  <p>{item.description}</p>
                                </Modal.Description>
                              </Modal.Content>
                            </Grid.Column>
                            <Grid.Column width={2} textAlign="center">
                              <Button
                                onClick={this.nextImage}
                                className="gallery noborder"
                                style={{ margin: 0 }}
                              >
                                <Icon
                                  name={aheadSVG}
                                  className="circled"
                                  size="30px"
                                  style={{ margin: 0 }}
                                />
                              </Button>
                            </Grid.Column>
                          </Grid.Row>
                        </Grid>
                      </Modal>
                    )}
                  </React.Fragment>
                ))}
            </Grid>
          </section>
        </article>
      </Container>
    );
  }
}

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
AlbumView.propTypes = {
  /**
   * Content of the object
   */
  content: PropTypes.shape({
    /**
     * Title of the object
     */
    title: PropTypes.string,
    /**
     * Description of the object
     */
    description: PropTypes.string,
    /**
     * Child items of the object
     */
    items: PropTypes.arrayOf(
      PropTypes.shape({
        /**
         * Title of the item
         */
        title: PropTypes.string,
        /**
         * Description of the item
         */
        description: PropTypes.string,
        /**
         * Url of the item
         */
        url: PropTypes.string,
        /**
         * Image of the item
         */
        image: PropTypes.object,
        /**
         * Image caption of the item
         */
        image_caption: PropTypes.string,
        /**
         * Type of the item
         */
        '@type': PropTypes.string,
      }),
    ),
  }).isRequired,
};

export default AlbumView;
