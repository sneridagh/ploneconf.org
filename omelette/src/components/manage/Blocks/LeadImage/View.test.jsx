import React from 'react';
import renderer from 'react-test-renderer';
import View from './View';

test('renders a view image component', () => {
  const component = renderer.create(
    <View
      data={{}}
      properties={{
        image: {
          download: 'image.png',
        },
      }}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
