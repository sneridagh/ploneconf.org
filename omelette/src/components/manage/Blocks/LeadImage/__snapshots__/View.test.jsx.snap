// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a view image component 1`] = `
<p
  className="block image align center"
>
  <img
    alt=""
    className=""
    src="image.png"
  />
</p>
`;
